worlds/sc2wol/PoolFilter.py

```diff
@@ -58,38 +59,32 @@ def filter_missions(multiworld: MultiWorld, player: int) -> Dict[str, List[str]]
-    # If the first mission is a build mission, there may not be enough locations to reach Outbreak as a second mission
+    # Excluding missions
+    for i, mission_pool in enumerate(mission_pools):
+        mission_pools[i] = [mission for mission in mission_pool if mission not in excluded_missions]
+    # Mission pool changes on Build-Only
     if not get_option_value(multiworld, player, 'shuffle_no_build'):
-        # Swapping Outbreak and The Great Train Robbery
-        if "Outbreak" in mission_pools[1]:
-            mission_pools[1].remove("Outbreak")
-            mission_pools[2].append("Outbreak")
-        if "The Great Train Robbery" in mission_pools[2]:
-            mission_pools[2].remove("The Great Train Robbery")
-            mission_pools[1].append("The Great Train Robbery")
-    # Removing random missions from each difficulty set in a cycle
-    set_cycle = 0
-    current_count = sum(len(mission_pool) for mission_pool in mission_pools)
-
-    if current_count < mission_count:
-        raise Exception("Not enough missions available to fill the campaign on current settings.  Please exclude fewer missions.")
-    while current_count > mission_count:
-        if set_cycle == 4:
-            set_cycle = 0
-        # Must contain at least one mission per set
-        mission_pool = mission_pools[set_cycle]
-        if len(mission_pool) <= 1:
-            if all(len(mission_pool) <= 1 for mission_pool in mission_pools):
-                raise Exception("Not enough missions available to fill the campaign on current settings.  Please exclude fewer missions.")
-        else:
-            mission_pool.remove(multiworld.random.choice(mission_pool))
-            current_count -= 1
-        set_cycle += 1
+        def move_mission(mission_name, current_pool, new_pool):
+            if mission_name in mission_pools[current_pool]:
+                mission_pools[current_pool].remove(mission_name)
+                mission_pools[new_pool].append(mission_name)
+        # Replacing No Build missions with Easy missions
+        move_mission("Zero Hour", 1, 0)
+        move_mission("Evacuation", 1, 0)
+        move_mission("Devil's Playground", 1, 0)
+        # Pushing Outbreak to Normal, as it cannot be placed as the second mission on Build-Only
+        move_mission("Outbreak", 1, 2)
+        # Pushing extra Normal missions to Easy
+        move_mission("The Great Train Robbery", 2, 1)
+        move_mission("Echoes of the Future", 2, 1)
+        move_mission("Cutthroat", 2, 1)
+        # Additional changes on Advanced Tactics
+        if get_option_value(multiworld, player, "required_tactics") > 0:
+            move_mission("The Great Train Robbery", 1, 0)
+            move_mission("Smash and Grab", 1, 0)
+            move_mission("Moebius Factor", 2, 1)
+            move_mission("Welcome to the Jungle", 2, 1)
+            move_mission("Engine of Destruction", 3, 2)
```

Comment:
I'd prefer at least constants than plain pool indexes for better readibility, something like  move_mission("Outbreak", POOL_EASY, POOL_NORMAL)

Reply:
Yeah, that might be nice. I'd recommend enum.IntEnum, which behaves like an int while offering typing checks.

Reply:
Condor's original code uses the strings "no_build", "easy", "medium", "hard", and "all_in" to identify mission difficulties, so I've changed PoolFilter to match the style of the code in MissionTables and Regions.  I would rather prioritize consistency in this case; I could change everything to an enum, but I don't want to tamper with Condor's original code design.

Reply:
If you feel up for the greater-scope refactor then I am giving it my thumbs up!